### ISSUE TITLE
Watched tasks do not terminate (sometimes)

### ISSUE THREAD

**Issue by nevir** [Microsoft/vscode#1312]
I'm finding that I have a bunch of zombie tasks, even after closing VS Code 0.10.3 (on OS X).

It appears that whenever I edit `tasks.json`, VS code forgets about the running tasks.  `> Tasks: Terminate Running Task` has no effect, and closing VS Code doesn't kill the process, either.

**Comment by dbaeumer** [comment]
@nevir can you provide a tasks.json file that shows the behavior. Usually VSCode should ask you to terminate the task when it closes.

**Comment by nevir** [comment]
Current config is below, but it doesn't seem to matter what the config is. Repro steps:

1. Start the build task (haven't tried it w/ named tasks)
2. Modify `tasks.json` (even just adding whitespace)
3. Observe that "terminate task" does nothing. Also, you can start up another copy of the task; double output will be routed to the output pane.

```json
{
  "version": "0.1.0",
  "command": "./node_modules/.bin/gulp",
  "isShellCommand": true,
  "showOutput": "silent",
  "tasks": [
    {
      "taskName": "watch",
      "isBuildCommand": true,
      "showOutput": "never",
      "isWatching": true,
      "problemMatcher": [
        {
          "owner": "eslint",
          "fileLocation": ["relative", "${workspaceRoot}"],
          "pattern": [
            {
              "regexp": "^\\s*(error)\\s+([\\w\\-]+)\\s+(.*)$",
              "severity": 1,
              "code": 2,
              "message": 3
            },
            {
              "regexp": "^\\s*(.*):(\\d+):(\\d+)$",
              "file": 1,
              "line": 2,
              "column": 3
            }
          ],
          "watching": {
            "beginsPattern": "^.*\\[wulp\\].+Running.*\\.\\.\\.$",
            "endsPattern": "^.*\\[wulp\\] Tasks ran in \\d+ms$"
          }
        }
      ]
    }
  ]
}
```

**Comment by dbaeumer** [comment]
@nevir This doesn't reproduce for me with a normal example. Would you be willing to share the project with us so that I can understand why this doesn't work for you?

**Comment by nevir** [comment]
Yeah, take a look at this repo: https://github.com/nevir/vscode-example-1

**Comment by dbaeumer** [comment]
@nevir thanks for the repository. I was able to reproduce the problem.

**Comment by dbaeumer** [comment]
@aeschli: to verify:

- clone https://github.com/Microsoft/vscode-languageserver-node.git
- cd client
- npm install
- open code
- Ctrl + Shift + B
- Add a compile error

Ensure that the spinner stops

**Comment by dbaeumer** [comment]
@aeschli ignore the steps from above. They should be:

- clone https://github.com/Microsoft/vscode-languageserver-node.git
- cd client
- npm install
- open code
- Ctrl + Shift + B
- Open tasks.json
- make a change (add a new line)
- Ctrl + Shift + B

Ensure that a message appears that there is already a running task. In the past a new task was started and two were running in parallel

**Comment by aeschli** [comment]
Verified in d740e5c4b

**Comment by kapilgorve** [comment]
I was facing this issue from 2-3 months. Today I just opened tasks.json and saw hint that isWatching is deprecated instead use isBackground. I changed it and ran the task and terminated ,Voila it's working.

I had to break my scss in error to stop the task when I wanted to exit VS Code.
